Patcher/UnityInstallation.cs

```diff
@@ -0,0 +1,98 @@
+namespace Patcher
+{
+    using System;
+    using System.Collections.Generic;
+    using System.IO;
+    using System.Text.RegularExpressions;
+
+    /// <summary>
+    /// Represents an Unity Installation on the Disk. 
+    /// </summary>
+    public class UnityInstallation
+    {
+        private readonly string _installationLocation;
+
+        private readonly OperatingSystem _operatingSystem;
+
+        /// <summary>
+        /// Creates a new Unity installation
+        /// </summary>
+        /// <param name="installationLocation">The Path to this specific installation on the File System</param>
+        /// <param name="operatingSystem"></param>
+        public UnityInstallation(string installationLocation, OperatingSystem operatingSystem)
+        {
+            _installationLocation = installationLocation;
```

Comment:
Since almost none of the API of this class would work when this is `null`, it'd probably make sense to add a precondition check along the lines of `_installationLocation = installationLocation ?? throw new ArgumentNullException(nameof(installLocation));`